lib/index.js

```diff
@@ -1,346 +1,61 @@
-     * controller with the following JSON object:
-     *
-     * {
-     *   controller: 'kuzzle-core-plugin-boilerplate/myNewController',
-     *   action: 'myNewAction',
-     *   ...
-     * }
-     *
-     * @type {Object}
-     *
-     * @see https://docs.kuzzle.io/plugins-reference/plugins-features/adding-controllers/
-     */
     this.controllers = {
       graphql: {
-        endpoint: 'graphQLEndpoint'
-      }
+        endpoint: 'graphQLEndpoint',
+      },
     };
 
-    /**
-     * The following "routes" exposed property allows Kuzzle to bind new
-     * controllers and actions to HTTP endpoints
-     *
-     * Any parameter starting with a ':' in the URL will be made dynamic by Kuzzle.
-     *
-     * The first route exposes the following GET URL:
-     *  https://<kuzzle server>:<port>/_plugin/kuzzle-core-plugin-boilerplate/say-something/<dynamic value>
-     *
-     * Kuzzle will call the function 'doSomething' with a Request object,
-     * containing the "name" property: request.input.args.property = '<dynamic value>'
-     *
-     * The second route exposes the following POST URL:
-     *  https://<kuzzle server>:<port>/_plugin/kuzzle-core-plugin-boilerplate/say-something
-     *
-     * Kuzzle will provide the content body of the request in the Request object
-     * passed to the function 'doSomething', in the request.input.body property
-     *
-     * @type {Array}
-     *
-     * @see https://docs.kuzzle.io/plugins-reference/plugins-features/adding-controllers/
-     */
     this.routes = [
       {
         verb: 'post',
         url: '/graphql',
         controller: 'graphql',
-        action: 'endpoint'
-      }
+        action: 'endpoint',
+      },
     ];
   }
 
-  /**
-   * Initializes the plugin with configuration and context.
-   *
-   * @param {Object} customConfig - This plugin custom configuration
-   * @param {Object} context      - A restricted gateway to the Kuzzle API
-   *
-   * @see https://docs.kuzzle.io/plugins-reference/plugins-creation-prerequisites/#plugin-init-function
-   * @see https://docs.kuzzle.io/plugins-reference/managing-plugins#configuring-plugins
-   * @see https://docs.kuzzle.io/plugins-reference/plugins-context/
-   */
   init(customConfig, context) {
-    this.config = Object.assign(this.config, customConfig);
+    this.config = customConfig;
     this.context = context;
+    const resolvers = new Resolvers(this.context).dump;
+    let typeDefs = {};
 
-    const typeDefs = fs
-      .readFileSync(path.join(__dirname, 'schema.gql'))
-      .toString();
-
-    const resolvers = {
-      Query: {
-        // drivers: () => {
-        //   // TODO
-        // },
-        driver: (previous, input, ctx, info) => {
-          return {
-            name: 'Sirkis',
-            license: 'B',
-            age: 49
-          };
-        }
-      }
-    };
+    try {
+      typeDefs = fs.readFileSync(path.join(__dirname, 'schema.gql')).toString();
```

Comment:
It's safer to always explicitly define the file encoding (and thus you don't need the call to `toString`) 
```suggestion
      typeDefs = fs.readFileSync(path.join(__dirname, 'schema.gql'), 'utf8');
```